src/main/java/uk/gov/hmcts/befta/player/DefaultBackEndFunctionalTestScenarioPlayer.java

```diff
@@ -621,12 +632,55 @@ private void authenticateUser(final BackEndFunctionalTestScenarioContext scenari
             UserData user) {
         String logPrefix = scenarioContext.getTestData().get_guid_() + ": " + prefix + " [" + user.getUsername() + "] ";
         String preferredTokenProviderClientId = scenarioContext.getTestData().getUserTokenClientId();
+
+        AuthenticationRetryConfiguration authenticationRetryConfiguration =
+                BeftaMain.getConfig().getAuthenticationRetryConfiguration();
+
         scenario.log("Authentication attempt from: " + preferredTokenProviderClientId + ".");
-        try {
+        if (authenticationRetryConfiguration.isRetryDisabled()) {
+            try {
+                BeftaMain.getAdapter().authenticate(user, preferredTokenProviderClientId);
+                BeftaUtils.defaultLog(logPrefix + "authenticated.");
+            } catch (Exception ex) {
+                throw new FunctionalTestException(logPrefix + "could not authenticate.", ex);
+            }
+        } else {
+            BeftaUtils.defaultLog("Authentication retry enabled");
+            authenticateUserWithRetry(authenticationRetryConfiguration, user, logPrefix, preferredTokenProviderClientId);
+        }
+    }
+
+    private void authenticateUserWithRetry(AuthenticationRetryConfiguration config,
+                                           UserData user,
+                                           String logPrefix,
+                                           String preferredTokenProviderClientId) {
+        AtomicInteger counter = new AtomicInteger(0);
+        Callable<Boolean> callable = () -> {
+            logger.info("User has been authenticated {} time(s).",counter.incrementAndGet());
             BeftaMain.getAdapter().authenticate(user, preferredTokenProviderClientId);
-            logger.info(logPrefix + "authenticated.");
-        } catch (Exception ex) {
-            throw new FunctionalTestException(logPrefix + "could not authenticate.", ex);
+            return true;
+        };
+
+        Retryer<Boolean> retryer = RetryerBuilder.<Boolean>newBuilder()
+                .retryIfResult(Predicates.isNull())
+                .retryIfExceptionOfType(IOException.class)
+                .retryIfExceptionOfType(ConnectException.class)
+                .retryIfRuntimeException()
+                .withStopStrategy(StopStrategies.stopAfterAttempt(config.getRetryAttempts()))
+                .withWaitStrategy(WaitStrategies.fibonacciWait(config.getRetryMultiplierTimeinMilliseconds(),
+                        config.getRetryMaxTimeInSeconds(),
+                        TimeUnit.SECONDS))
+                .build();
+        try {
+            retryer.call(callable);
+            BeftaUtils.defaultLog(String.format("%s authenticated.", logPrefix));
```

Comment:
This log line doesn't do what is actually asked. 
"Could we add a log line to tell how many-th time this is an attempt to authenticate?"